ArduCopter/mode_auto.cpp

```diff
@@ -802,6 +815,7 @@ void ModeAuto::takeoff_run()
     if ((copter.g2.auto_options & (int32_t)Options::AllowTakeOffWithoutRaisingThrottle) != 0) {
         copter.set_auto_armed(true);
     }
+    wp_nav->shift_wp_origin_and_destination_to_current_pos_xy();
```

Comment:
we shouldn't need this addition any more I think because we're not using wp_nav for auto takeoffs anymore..